test/libyul/yulOptimizerTests/fullSuite/create2_and_mask.yul

```diff
@@ -0,0 +1,21 @@
+{
+    // This does not optimize the masks away. Due to the way the expression simplifier
+    // is built, it would have to create another `create2` opcode for the simplification
+    // which wolud be fatal.
+    let a := and(create2(0, 0, 0x20, 0), 0xffffffffffffffffffffffffffffffffffffffff)
+    let b := and(0xffffffffffffffffffffffffffffffffffffffff, create2(0, 0, 0x20, 0))
+    sstore(a, b)
+}
+// ====
+// EVMVersion: >=constantinople
+// ----
+// step: fullSuite
+//
+// {
+//     {
+//         let _1 := sub(shl(160, 1), 1)
+//         let _2 := 0
+//         let a := and(create2(_2, _2, 0x20, _2), _1)
+//         sstore(a, and(_1, create2(_2, _2, 0x20, _2)))
```

Comment:
For develop this would be
```solidity
            let _1 := 0xffffffffffffffffffffffffffffffffffffffff
            let _2 := 0
            let a := and(create2(_2, _2, 0x20, _2), _1)
            sstore(a, and(_1, create2(_2, _2, 0x20, _2)))
```

Why is this different now?

Reply:
I just ran it on develop and do not see a difference.

Reply:
I just ran it using ` --strict-assembly --ir-optimized --optimize`. Let me try by adding it as a test in `yulOptimizerTests/fullsuite/`

Reply:
I checked by adding a test to `yulOptimizerTests/fullsuite/`, and the output is the same. Didn't expect that the output would be different from running `solc`.

Reply:
solc also runs the legacy optimizer